src/Core/Utilities/DuoApi.cs

```diff
@@ -175,22 +175,21 @@ public T JSONApiCall<T>(string method, string path, Dictionary<string, string> p
             var res = ApiCall(method, path, parameters, timeout, out var statusCode);
             try
             {
+                // TODO: We should deserialize this into our own DTO and not work on dictionaries.
                 var dict = JsonSerializer.Deserialize<Dictionary<string, object>>(res);
```

Comment:
I would recommend using a `JsonDocument` here and working with that to get values in the type that you want them.